Service points added

Smetje opened the pull request:
Added the possibility to request service points for a given area. 
The area is defined by NorthEast latitude and longitude and the SouthWest latitude and longitude.

Smetje commented:
Also added fillables for Parcel

jschram commented on src/Picqer/Carriers/SendCloud/Connection.php:
```diff
@@ -112,12 +112,16 @@ public function apiUrl()
     {
         try {
+            if ($auth) {
+                $params[$auth] = $this->apiKey;
```
Shouldn't this be `$params['auth']` instead? Now it'll expand to `$params[true]`

Smetje replied:
Nope, to leave all possibilities open, the $auth param could be filled with whatever string you want. 
For example to fetch the service points the auth parameter is filled in with 'access_token':
$result = $this->connection->get($url, $params, 'access_token');

jschram replied:
In that case, maybe it makes more sense to do `if (!is_null($auth)) {` instead and set the default value for `$auth` to `null` instead. Seems a bit more readable than mixing boolean and strings IMHO.

Smetje replied:
Makes sense, I've changed it.

casperbakker commented:
Hi Smetje,

I am sorry, but I don't think this is a proper implementation. It is not build like the rest of this client and also that 'access_token' method needs to much explaining, meaning it is not clear enough.

Things I think need to be changed:
- The $fillable of ServicePoint should have all the items that SendCloud returns for a resource. For service points that is id, name, street, latitude, etc...
- We do not use `DS` in our code, so please just insert ` . '/' . ` with spacing in between for the URL
- Do not use that 'auth' parameter, it is too hacky. The thing you want to achieve can be done by just adding the apiKey in the params array in the get method of the ServicePoint
- You can get bonus points if that get() method is also in a trait and you use that trait in the ServicePoint model

And personally I think there is a better name then 'get()'. We already have an 'all()', and this is like the all but filtered. I don't think there should be a get() and an all(). If it cannot be merged into one, it should have a better name in my opinion.

Smetje commented:
Hi casperbakker,

Thanks for the feedback. I'm not familiar with traits, sorry about not using these properly.

Connection.php is cleaned up. I've added a function to get the private parameter 'apiKey', hope this is ok? It was the only way to get the apiKey in a trait.

i've created a trait 'Filter', used by ServicePoint.php only. Not included in 'Findable.php'. And used an 'all()'. 

ServicePoint.php is cleaned up and fillables are correct now.

Smetje commented:
Can you give some more information about $namespaces?
With ServicePoints you don't have a namespace. Can it be removed from the object as there is a default in Model.php?

casperbakker commented:
Hi Smetje,

Not a problem you are not familiar with them, open source projects are a great way to learn new things.

Public method for the API key is fine. It is weird of SendCloud to not also support apikey/secret in the normal way also. It would be better to call that function 'getApiKey()', to make it more clear what it means.

The trait is good, but I still think there should be an other name then get/all for the filter. Now you cannot use the Query and Filter trait in one model. That is a situation that can occur in the future.

You can fix that, or otherwise I will do some tweaking before merging.

Thanks for your contribution.

Smetje commented:
Hi casperbakker,

I've changed the apiKey function name to getApiKey and changed it in Filter.php.

Can't come up with another name for get/all, so I leave that up to you.

Smetje commented:
What about using 'Filter' as name for the function?

smetje commented:
Up
